service/operator/service.go

```diff
@@ -0,0 +1,138 @@
+type Config struct {
+	// Dependencies.
+	Logger    micrologger.Logger
+	K8sClient kubernetes.Interface
+
+	// Settings.
+	Flag  *flag.Flag
+	Viper *viper.Viper
+}
+
+// DefaultConfig provides a default configuration to create a new operator
+// service by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		K8sClient: nil,
+		Logger:    nil,
+
+		// Settings.
+		Flag:  nil,
+		Viper: nil,
+	}
+}
+
+// Service implements the Operator service interface.
+type Service struct {
+	Config
+
+	// Internals.
+	bootOnce sync.Once
+	tpr      *tpr.TPR
+}
+
+// New creates a new configured Operator service.
+func New(config Config) (*Service, error) {
+	// Dependencies.
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "kubernetes client must not be empty")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "logger must not be empty")
+	}
+
+	// Settings.
+	if config.Flag == nil {
+		return nil, microerror.Maskf(invalidConfigError, "flag must not be empty")
+	}
+	if config.Viper == nil {
+		return nil, microerror.Maskf(invalidConfigError, "viper must not be empty")
+	}
+
+	tprConfig := tpr.Config{
+		K8sClient:   config.K8sClient,
+		Logger:      config.Logger,
+		Name:        azuretpr.Name,
+		Version:     TPRVersion,
+		Description: TPRDescription,
+	}
+	tpr, err := tpr.New(tprConfig)
+	if err != nil {
+		return nil, microerror.Maskf(err, "creating TPR for %#v", tprConfig)
+	}
+
+	newService := &Service{
+		Config: config,
+
+		// Internals
+		bootOnce: sync.Once{},
+		tpr:      tpr,
+	}
+
+	return newService, nil
+}
+
+// Boot starts the service and implements the watch for azuretpr.
+func (s *Service) Boot() {
+	s.bootOnce.Do(func() {
+		err := s.tpr.CreateAndWait()
+		if tpr.IsAlreadyExists(err) {
+			s.Logger.Log("debug", "third party resource already exists")
+		} else if err != nil {
+			s.Logger.Log("error", fmt.Sprintf("%#v", err))
+			return
+		}
+
+		s.Logger.Log("debug", "starting list/watch")
+
+		newResourceEventHandler := &cache.ResourceEventHandlerFuncs{
+			AddFunc:    s.addFunc,
+			DeleteFunc: s.deleteFunc,
+		}
+		newZeroObjectFactory := &tpr.ZeroObjectFactoryFuncs{
+			NewObjectFunc:     func() runtime.Object { return &azuretpr.CustomObject{} },
+			NewObjectListFunc: func() runtime.Object { return &azuretpr.List{} },
+		}
+
+		s.tpr.NewInformer(newResourceEventHandler, newZeroObjectFactory).Run(nil)
+	})
+}
+
+func (s *Service) addFunc(obj interface{}) {
+	customObject := *obj.(*azuretpr.CustomObject)
+	s.Config.Logger.Log("debug", fmt.Sprintf("creating cluster '%s'", customObject.Spec.Cluster.Cluster.ID))
+
+	// TODO Add stub code for creating an Azure Resource Group.
+}
+
+// deleteFunc TODO
+func (s *Service) deleteFunc(obj interface{}) {
+	customObject := *obj.(*azuretpr.CustomObject)
+	s.Config.Logger.Log("debug", fmt.Sprintf("deleting cluster '%s'", customObject.Spec.Cluster.Cluster.ID))
```

Comment:
`Config` is embedded you can just call `s.Logger.Log`

When this is merged https://github.com/giantswarm/micrologger/commit/3fc80a5b178fd25a757c8bbe54599b21b11a1110 you can now do something like this:

```
logger := s.Logger.With("cluster", customObject.Spec.Cluster.Cluster.ID)

logger.Log("debug", "deleting cluster)

....

logger.Log(...)
```